rewrite-java/src/main/java/org/openrewrite/java/cleanup/UseGenericTypeMethodParameters.java

```diff
@@ -0,0 +1,257 @@
+        return "Use type parameters instead of raw types";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replaces raw method parameters on private methods with an inferable generic type. " +
+                "The raw type will not be updated if a raw type or more than 1 type is passed to the method through method invocations in the class. " +
+                "By default, the recipe will not update methods with any annotations on a parameter. I.E. `@Autowired Object arg0`";
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("RSPEC-3740");
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(2);
+    }
+
+    public static Set<String> rspecRawTypes = new HashSet<>();
+
+    static {
+        rspecRawTypes.add("java.util.Deque");
+        rspecRawTypes.add("java.util.List");
+        rspecRawTypes.add("java.util.Queue");
+        rspecRawTypes.add("java.util.Set");
+    }
+
+    @Override
+    protected JavaVisitor<ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+
+            @Override
+            public JavaSourceFile visitJavaSourceFile(JavaSourceFile cu, ExecutionContext executionContext) {
+                for (JavaType type : cu.getTypesInUse().getTypesInUse()) {
+                    JavaType.FullyQualified fq = TypeUtils.asFullyQualified(type);
+                    if (fq != null && rspecRawTypes.contains(fq.getFullyQualifiedName())) {
+                        return super.visitJavaSourceFile(cu, executionContext);
+                    }
+                }
+                return cu;
+            }
+
+            @Override
+            public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext executionContext) {
+                J.MethodDeclaration m = super.visitMethodDeclaration(method, executionContext);
+                if (method.hasModifier(J.Modifier.Type.Private) && !isOverload(method.getSimpleName()) &&
+                        !method.isConstructor() && containsTargetParamType(method)) {
+
+                    Map<Integer, Set<JavaType>> parameterizedArguments = compareMethodInvocations(method);
+                    if (parameterizedArguments.isEmpty()) {
+                        return m;
+                    }
+
+                    // Do not transform the method if any of the invocations passed in a raw type or there is more than 1 type associated to the parameter.
+                    if (parameterizedArguments.values().stream().anyMatch(o -> o.size() > 1 ||
+                            o.stream().anyMatch(type -> type instanceof JavaType.Unknown))) {
+                        return m;
+                    }
+
+                    Optional<String> parameterCode = generateTemplateCode(method.getParameters(), parameterizedArguments);
+                    if (parameterCode.isPresent()) {
```

Comment:
Todo:  this needs to update each of the identifier types.
Add a visitor to update each of the raw types to the parameterized type within the method dec block.